### PR TITLE
Migrate test-app to glimmer components and newer ember syntax

### PR DESCRIPTION
**Pull request by m-jovan**
We're freshening up the test-app with migrating components to glimmer and routes to classes. With this we are able to have clearer syntax and unblock future updates of docs libs and ember.

Also this lets us continue with the typescript migration.

The PR is best reviewed commit by commit

This closes https://github.com/qonto/ember-phone-input/pull/726 since it installs the package and fixes failing actions.

### CODE REVIEW

**Comment by m-jovan** [comment]
#web

File: test-app/app/components/docs/components/phone-input/all-options.hbs [hunk 1]
```diff
@@ -0,0 +1,129 @@
+<h2>Basic Options</h2>
```
Comment (nicolasgasco) [inline test-app/app/components/docs/components/phone-input/all-options.hbs:1]: Where can I see the rendered output of this?

File: test-app/app/components/docs/components/phone-input/all-options.hbs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        @initialCountry='us'
+        @number={{this.number}}
+        @update={{this.handleUpdate}}
+      /></p>
+
+    <p>The internal phone number you'll want to persist on your backend is
+      <em>\{{this.number}}: </em><strong>{{this.number}}</strong>.</p>
+
+    <p>In some cases, you may want to persist some additional metadata. Are
+      available:</p>
```
Comment (nicolasgasco) [inline test-app/app/components/docs/components/phone-input/all-options.hbs:15]: I find "Are available" not very grammatical. Maybe you can rephrase it to something like: "Following properties are available:"?

**Review by nicolasgasco — APPROVED** [review]

File: test-app/app/components/docs/components/phone-input/all-options.hbs (hunk 1 above)
Comment (m-jovan) [reply test-app/app/components/docs/components/phone-input/all-options.hbs:1]: That's the docs page that can be found on localhost 4202 locally or on github pages: https://qonto.github.io/ember-phone-input/versions/master/

File: test-app/app/components/docs/components/phone-input/all-options.hbs (hunk 2 above)
Comment (m-jovan) [reply test-app/app/components/docs/components/phone-input/all-options.hbs:15]: We are going to replace docs with markdown in the next PR and also I'd really like to avoid mixing docs changes with the typescript migration that is done in this PR if that's ok. WDYT?

File: test-app/app/components/docs/components/phone-input/action-handling.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+  @tracked metaData;
+
+  @action
+  // BEGIN-SNIPPET phone-input-action-handling.js
+  handleUpdate(number, metaData) {
+    this.number = number;
+    this.metaData = metaData;
   }
-});
+  // END-SNIPPET
```
Comment (vscav) [inline test-app/app/components/docs/components/phone-input/action-handling.js:15]: What are these comments for?

File: test-app/app/components/docs/components/phone-input/all-options.hbs (hunk 2 above)
Comment (nicolasgasco) [reply test-app/app/components/docs/components/phone-input/all-options.hbs:15]: Absolutely OK, yes!

File: test-app/app/components/docs/components/phone-input/action-handling.js (hunk 3 above)
Comment (m-jovan) [reply test-app/app/components/docs/components/phone-input/action-handling.js:15]: I believe for linking the snippet to docs page

**Review by vscav — APPROVED** [review]

File: test-app/app/templates/docs/lazy-loading.md [hunk 4]
```diff
@@ -22,11 +22,11 @@ module.exports = function(environment) {
...
 ```js
-export default Route.extend({
-  phoneInput: service(),
+export default class Route extends Route {
+  @service phoneInput;
 
-  beforeModel() {
-    return this.get('phoneInput').load()
+  async beforeModel() {
+    await this.phoneInput.load();
```
Comment (dannycalleri) [inline test-app/app/templates/docs/lazy-loading.md:29]: Are we missing a `return` here? Maybe this could be:

```suggestion
  async beforeModel() {
    return await this.phoneInput.load();
```

or maybe even better:

```suggestion
  async beforeModel() {
    return this.phoneInput.load();
```

Comment (m-jovan) [reply test-app/app/templates/docs/lazy-loading.md:29]: No we actually just want to lazy load (import dynamically) everything within the service so the component can use it later internally

Comment (dannycalleri) [reply test-app/app/templates/docs/lazy-loading.md:29]: Gotcha thanks!

**Review by dannycalleri — APPROVED** [review]
